internal/common/objects/Audi.go

```diff
@@ -0,0 +1,72 @@
+package objects
+
+import (
+	phy "SOMAS2023/internal/common/physics"
+	"SOMAS2023/internal/common/utils"
+	"math"
+)
+
+type IAudi interface {
+	IBaseBiker
+	IPhysicsObject
+}
+
+type Audi struct {
+	*BaseBiker
+	*PhysicsObject
+	target *MegaBike
+}
+
+// GetAudi is a constructor for Audi that initializes it with a new UUID and default position.
+func GetAudi() *Audi {
+	return &Audi{
+		PhysicsObject: GetPhysicsObject(utils.MassAudi),
+	}
+}
+
+func GetIAudi() IAudi {
+	return &Audi{
+		PhysicsObject: GetPhysicsObject(utils.MassAudi),
+	}
+}
+
+// Move as MegaBike, called by server
+func (audi *Audi) Move() {
+	audi.coordinates = phy.GetNewPosition(audi.coordinates, audi.velocity, audi.orientation)
+}
+
+// DecideForce is called by server as an agent, Audi finds possible target and move towards it at constant velocity
+func (audi *Audi) DecideForce() {
+	// the speed of audi should be constant, use force 1 to represent
+	if audi.target == nil { // no target, audi will stop
+		audi.velocity = 0.0
+	} else {
+		audi.velocity = 1.0 / audi.mass
```

Comment:
Can you replace audi.mass with utils.MassAudi (in CommonParameters.go)?

Reply:
audi.mass equals to utils.MessAudi when Audi was initialized. I think there is no difference for audi.mass or utils.MassAudi here.

Reply:
My main worry is with the 'fairness' of the simulation:

Agents move by supplying a force, which induces an acceleration, which induces a velocity. This means that the velocity isn't instantaneous. 

Instantaneous velocity here corresponds to an infinite force, meaning that this Audi doesn't obey the same physics that the agents do.

I think that the "DecideForce" from the agent code is imperative here, otherwise the Audi will just 'teleport' around

Reply:
The force can be regarded as 1.0 constantly, we can simply change the MassAudi config to change the speed of chasing. The velocity is calculated as Force/Mass which is actually following the physics law defined in engine.

Reply:
So as to make the changes to Engine clearer, I will address this in a new PR - EngineUpdate